modules/ui/fb_feature_picker.js

```diff
@@ -13,36 +13,37 @@ import { uiRapidFirstEdit } from './rapid_first_edit_dialog';
 
 export function uiFbFeaturePicker(context, keybinding) {
     var _datum;
-    var ML_ROADS_LIMIT_NON_TM_MODE = 50;
+    var AI_FEATURES_LIMIT_NON_TM_MODE = 50;
 
 
-    function isAddRoadDisabled() {
+    function isAddFeatureDisabled() {
         // when task GPX is set in URL (TM mode), "add roads" is always enabled
         var gpxInUrl = utilStringQs(window.location.hash).gpx;
         if (gpxInUrl) return false;
 
-        var mlRoadsCount = 0;
-        var entities = context.graph().entities;
-        for (var eid in entities) {
-            var e = entities[eid];
-            if (eid.startsWith('w-') && e && (e.tags.source === 'digitalglobe' || e.tags.source === 'maxar')) {
-                mlRoadsCount += 1;
+        var aiFeatures = context.history().peekAllAnnotations(); 
+
+        var aiFeatureCount = 0; 
+        for (var i = 0; i <= aiFeatures.length; i++) {
+            if (aiFeatures[i] && aiFeatures[i].type === 'fb_accept_feature') {
+                aiFeatureCount++; 
             }
         }
-        return mlRoadsCount >= ML_ROADS_LIMIT_NON_TM_MODE;
+        
+        return aiFeatureCount >= AI_FEATURES_LIMIT_NON_TM_MODE;
```

Comment:
Annotations are guaranteed to exist from the core/history.js method, so you can safely drop the check for `aiFeatures[i]` in `if (aiFeatures[i] && ...)`.

Also, it's effectively a filter & count, so if you want to get crazy:

```
var aiFeatureAccepts = annotations.filter(function (a) { return a.type === 'fb_accept_feature'; });
return aiFeatureAccepts.length >= AI_FEATURES_LIMIT_NON_TM_MODE;
```